ui/src/app/components/admin/audit-card-list-view.tsx

```diff
@@ -193,17 +210,109 @@ const AuditActionCard = (props: { action: AuditAction }) => {
   );
 };
 
-export const AuditActionCardListView = (props: { actions: AuditAction[]}) => {
+interface FilterEntry {
+  displayName?: string;
+  isActive: boolean;
+}
+
+interface FilterState {
+  [key: string]: FilterEntry;
+}
+
+const ActionTypeFilter = (props: {
+  activeActionTypes: FilterState,
+  updateFilter: (actionType: string, isActive: boolean) => void }) => {
+
+  const {activeActionTypes, updateFilter} = props;
+
+  const toggleSelectedAction = (actionType) => {
+    updateFilter(actionType, !fp.get(`${actionType}.isActive`)(activeActionTypes));
+  };
+
+  return (
+      <React.Fragment>
+        <div style={{
+          marginLeft: '0',
+          padding: '2px',
+          display: 'flex',
+          flexDirection: 'column',
+          textAlign: 'left',
+          border: `1px solid ${colors.secondary}`,
+          width: 'fit-content'
+        }}>
+        <div style={{fontWeight: 600, color: colors.accent}}>Action Types</div>
+          {
+            fp.flow(
+              fp.toPairs,
+              fp.toPairs,
+              fp.map(([index, [id, entry]]: [number, [string, FilterEntry]]) =>
+                    <CheckBox
+                        key={index}
+                        id={`${id}_active`}
+                        onChange={() => toggleSelectedAction(id)}
+                        label={entry.displayName}
+                        checked={entry.isActive}
+                        style={{margin: '0.25rem'}}/>)
+            )(activeActionTypes)
+          }
+        </div>
+    </React.Fragment>);
+};
+
+export const AuditActionCardListView = (props: { actions:  AuditAction[]}) => {
   const {actions} = props;
+  const [activeActionTypes, setActiveActionTypes] = useState({});
+
+  const getActionTypes = (action: AuditAction) => fp.flow(
+    fp.map((e: AuditEventBundle) => e.header.actionType),
+    fp.uniq
+  )(action.eventBundles);
+
+  useEffect(() => {
+    const actionTypeToShow = fp.flow(
+      fp.flatMap(getActionTypes),
+      fp.countBy(fp.identity),
+      fp.toPairs,
+      fp.map(([actionName, count]: [string, number]) => [actionName, {
+        isActive: true,
+        displayName: `${toTitleCase(actionName)} (${count || 0})`
+      }]),
+      fp.fromPairs
+    )(actions);
+
+    setActiveActionTypes(actionTypeToShow);
+  }, [actions]);
+
+  const updateFilterCallback = (actionType: string, isActive: boolean) => {
+    const newActiveActionTypes = fp.clone(activeActionTypes);
```

Comment:
Is there a slicker way to do this? In Java, there's a builder pattern that will let you do `Builder.from(originalObj).withId(2).build` with no  mutations.

Reply:
Missed this comment when I approved - let me take a look

Reply:
It's just a curiosity, so no worries.

Reply:
I saw you merged, but if you are looking for an alternative this should work:
```javascript
 const updateFilterCallback = (actionType: string, isActive: boolean) => {
    setActiveActionTypes(fp.set([actionType, 'isActive'], isActive, activeActionTypes));
  };
```